mkfs.jffs2.c: fix device_table handling

find_filesystem_entry would ignore device_table entries for existing elements
if the device type / permissions weren't exactly the same, effectively
rendering device_table useless.
---
 mkfs.jffs2.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

Index: mtd-utils-1.1.0/mkfs.jffs2.c
===================================================================
--- mtd-utils-1.1.0.orig/mkfs.jffs2.c
+++ mtd-utils-1.1.0/mkfs.jffs2.c
@@ -246,8 +246,7 @@
 		e = dir->files;
 	}
 	while (e) {
-		/* Only bother to do the expensive strcmp on matching file types */
-		if (type == (e->sb.st_mode & S_IFMT)) {
+		if (1) {
 			if (S_ISDIR(e->sb.st_mode)) {
 				int len = strlen(e->fullname);
 
